daemon/tests/happy_path.rs

```diff
@@ -59,6 +64,150 @@ async fn taker_takes_order_and_maker_rejects() {
     assert!(matches!(maker_cfd.state, CfdState::Rejected { .. }));
 }
 
+#[tokio::test]
+async fn taker_proposes_rollover_and_maker_accepts_rollover() {
+    let _guard = init_tracing();
+    let (mut maker, mut taker) = start_both().await;
+
+    maker.mocks.mock_oracle_annoucement().await;
+    taker.mocks.mock_oracle_annoucement().await;
+
+    maker.mocks.mock_party_params().await;
+    taker.mocks.mock_party_params().await;
+
+    maker.mocks.mock_oracle_attestation().await;
+    taker.mocks.mock_oracle_attestation().await;
+
+    maker.mocks.mock_monitor_attestation().await;
+    taker.mocks.mock_monitor_attestation().await;
+
+    maker.mocks.mock_start_monitoring().await;
+    taker.mocks.mock_start_monitoring().await;
+
+    is_next_none(&mut taker.order_feed).await;
+
+    maker.publish_order(dummy_new_order()).await;
+
+    let (_, received) = next_order(&mut maker.order_feed, &mut taker.order_feed).await;
+
+    taker.take_order(received.clone(), Usd::new(dec!(5))).await;
+    let (_, _) = next_cfd(&mut taker.cfd_feed, &mut maker.cfd_feed).await;
+
+    maker.accept_take_request(received.clone()).await;
+
+    let (taker_cfd, maker_cfd) = next_cfd(&mut taker.cfd_feed, &mut maker.cfd_feed).await;
+    // TODO: More elaborate Cfd assertions
+    assert_eq!(taker_cfd.order.id, received.id);
+    assert_eq!(maker_cfd.order.id, received.id);
+    assert!(matches!(taker_cfd.state, CfdState::ContractSetup { .. }));
+    assert!(matches!(maker_cfd.state, CfdState::ContractSetup { .. }));
+
+    mock_wallet_sign_and_broadcast(&mut maker.mocks.wallet().await);
+    mock_wallet_sign_and_broadcast(&mut taker.mocks.wallet().await);
+
+    let (taker_cfd, maker_cfd) = next_cfd(&mut taker.cfd_feed, &mut maker.cfd_feed).await;
+
+    // TODO: More elaborate Cfd assertions
+    assert_eq!(taker_cfd.order.id, received.id);
+    assert_eq!(maker_cfd.order.id, received.id);
+    assert!(matches!(taker_cfd.state, CfdState::PendingOpen { .. }));
+    assert!(matches!(maker_cfd.state, CfdState::PendingOpen { .. }));
+
+    maker
+        .cfd_actor_addr
+        .send(monitor::Event::LockFinality(maker_cfd.order.id))
+        .await
+        .unwrap();
+    taker
+        .cfd_actor_addr
+        .send(monitor::Event::LockFinality(taker_cfd.order.id))
+        .await
+        .unwrap();
+
+    let (taker_cfd_open_state, maker_cfd_open_state) =
+        next_cfd(&mut taker.cfd_feed, &mut maker.cfd_feed).await;
+
+    assert!(matches!(taker_cfd_open_state.state, CfdState::Open { .. }));
+    assert!(matches!(maker_cfd_open_state.state, CfdState::Open { .. }));
```

Comment:
All of this is essentially setup code, at least from the perspective of this test. Please create a helper like `start_both_in_open` that does all of this minus the assertions (setup code should not contain assertions).

Reply:
I am not sure how to do this. in order to have setup and assertions seperate I would have to return multiple states from the setup function and assert on them inside the test. I think this might be a bit confusing.

Reply:
I think it is fine to have setup functions that don't have any assertions. We have a separate test that has detailed assertions things that are meant to happen on the way to an open CFD.

We can have a setup function that assumes that this works and just creates an open CFD. If anything breaks within that part of the code base, all these tests will fail (and probably with different errors). It is then up to the dev to check which test is the one that points out the problem with the most detail and which tests are just a symptom. That should be easy. If something panics in the setup, no need to investigate that test.

Reply:
out of curiosity what is the problem in leaving the assertions in the setup functions. Wont it make it easier to debug in the case something goes wrong in the future?